src/api/app/models/project.rb

```diff
@@ -58,7 +58,10 @@ def autocomplete(search)
   has_many :messages, :as => :db_object, :dependent => :delete_all
   has_many :watched_projects, :dependent => :destroy, inverse_of: :project
 
-  has_many :linkedprojects, -> { order(:position) }, :class_name => 'LinkedProject', foreign_key: :db_project_id, :dependent => :delete_all
+  has_many :linked_projects, -> { order(:position) }, foreign_key: :db_project_id, :dependent => :delete_all
+  has_many :projects_linked, through: :linked_projects, class_name: 'Project', source: :linked_db_project
+  has_many :linking_projects, -> { order(:position) }, :class_name => 'LinkedProject', foreign_key: :linked_db_project_id, :dependent => :delete_all
+  has_many :projects_linking, through: :linking_projects, class_name: 'Project', source: :project
```

Comment:
Huh, that's very confusing:) So a project can have a relation (link) to a LinkedProject, right? What are the other 3 relations pointing to?